### ISSUE TITLE
Problems with `create!` with validations

### ISSUE THREAD

**Issue by ProGM** [neo4jrb/neo4j#1182]
Hi there,
I think I found a bug with `create!` method. It seems not to accept relations as parameters. Instead, if I use `new(params).save!` It works well

Here's an example I tried:

```ruby
class Article
  include Neo4j::ActiveModel
  property :title
  has_one :out, :user, type: :author
  validates :user, presence: true
end

Article.create!(title: 'test', user: User.first)
# => Neo4j::ActiveNode::Persistence::RecordInvalidError: User can't be blank

Article.new(title: 'test', user: User.first).save!
# => true
```

### Runtime information:

Neo4j database version: 2.2.7
`neo4j` gem version: 7.0.0
`neo4j-core` gem version: 6.1.2

**Comment by subvertallchris** [comment]
I'm on my phone, sitting on a beach in Mexico right now so I'm having a
little trouble linking to the line in code but it looks like we're handling
associations different for `save` and `create`. `save` is totally capable
of handling associations but for some reason, `create` pulls those keys out
first, then calls
`save`, then assigns in another operation. I only glanced at it very
quickly but it looks like we should be able to redefine `create` as:

```ruby

def create(props)
  new(props).tap { |node| node.save }
end

```

With the ! method defined and calling as expected, too.

**Comment by subvertallchris** [comment]
Since this was a bug, I just released the fix as 7.0.4. :-)
